src/basicfuns.jl

```diff
@@ -441,3 +441,72 @@ $(SIGNATURES)
 Compute the complementary double exponential, `1 - exp(-exp(x))`.
 """
 cexpexp(x) = -_expm1(-exp(x))
+
+#=
+this uses the identity:
+
+log(logistic(x)) = -log(1 + exp(-x))
+=#
+"""
+$(SIGNATURES)
+
+The natural logarithm of the `logistic` function, computed more
+carefully and with fewer calls than than the composition
+`log(logistic(x))`.
+
+Its inverse is the [`logitexp`](@ref) function.
+"""
+loglogistic(x::AbstractFloat) = -log1pexp(-x) #
+loglogistic(x::T) where {T<:Real} = -log1pexp(-convert(promote_type(Float64, T), x))
+
+#=
+this uses the identity:
+
+logit(exp(x)) = log(exp(x) / (1 + exp(x))) = log(exp(x)) - log(1 - exp(x))
+=#
+"""
+$(SIGNATURES)
+
+The logit of the exponential of `x`, computed more carefully and
+with fewer function calls than `logit(exp(x))`
+
+Its inverse is the [`loglogistic`](@ref) function.
+"""
+logitexp(x::Real) = x - log1mexp(x)
```

Comment:
Should this be done more carefully to avoid loss of precision when both terms are approximately equal?

Reply:
You're right -- this can be done more carefully using `logexpm1`.
However, as illustrated in the figures, `logexpm1` itself suffers a loss of precision around `-log(2)`.

Absolute error is ok.
![logitexp_abserr](https://github.com/JuliaStats/LogExpFunctions.jl/assets/96091198/4e2f93b5-137d-495e-9998-b28a3b578fe3)

But units in the last place error is a problem around `-log(2)`
![logitexp_ulperr](https://github.com/JuliaStats/LogExpFunctions.jl/assets/96091198/2a346db0-bfc9-4754-9105-bdfa41402efe)

Reply:
As this can only be resolved by improving the ulp error of `logexpm1` around `log(2)`, I see no reason for this to gate merging of this PR.

Reply:
I think we could check if the argument is in a neighborhood of -log(2), and then just use a first-order approximation. My quick calculations suggest that the local derivative is 2 but someone should check this.